### PR TITLE
Support Rails 6 asset recompilation through webpacker 

### PR DESCRIPTION
**Pull request by the-spectator**
In Rails 6, Webpacker is the default JavaScript compiler. 
This PR adds app/javascript/ to asset_dir & public/packs/ to complied_asset_path as it is the default directory used by Webpacker.

### CODE REVIEW

**Comment by the-spectator** [comment]
Hi @d4be4st, just a friendly ping. Can you please check this PR.

**Comment by varunempire** [comment]
@rstacruz Can you please check this PR..

**Comment by m4c3** [comment]
Can you please merge this?

File: tasks/mina/rails.rb [hunk 1]
```diff
@@ -5,11 +5,11 @@
 set :bundle_prefix, -> { %{RAILS_ENV="#{fetch(:rails_env)}" #{fetch(:bundle_bin)} exec} }
 set :rake, -> { "#{fetch(:bundle_prefix)} rake" }
 set :rails, -> { "#{fetch(:bundle_prefix)} rails" }
-set :compiled_asset_path, 'public/assets'
-set :asset_dirs, ['vendor/assets/', 'app/assets/']
+set :compiled_asset_path, ['public/assets', 'public/packs']
+set :asset_dirs, ['vendor/assets/', 'app/assets/', 'app/javascript/']
```
Comment (lovro-bikic) [inline tasks/mina/rails.rb:9]:
```suggestion
set :asset_dirs, ['vendor/assets/', 'app/assets/', 'app/javascript/', 'package.json']
```

Assets should recompile too when dependencies in package.json change.

File: tasks/mina/rails.rb [hunk 2]
```diff
@@ -5,11 +5,11 @@
...
 set :migration_dirs, ['db/migrate']
 
-set :shared_dirs, fetch(:shared_dirs, []).push('log', 'tmp/cache', fetch(:compiled_asset_path))
+set :shared_dirs, fetch(:shared_dirs, []).push('log', 'tmp/cache', *fetch(:compiled_asset_path))
```
Comment (lovro-bikic) [inline tasks/mina/rails.rb:12]:
```suggestion
set :shared_dirs, fetch(:shared_dirs, []).push('log', 'tmp/cache', 'node_modules', *fetch(:compiled_asset_path))
```

**Review by lovro-bikic — CHANGES_REQUESTED** [review]
Hi, thanks for the PR. It's good overall, I'd just like to add `package.json` to asset dirs and `node_modules` to shared dirs.

File: tasks/mina/rails.rb (hunk 2 above)
Comment (the-spectator) [reply tasks/mina/rails.rb]: @lovro-bikic Applied suggested changes.

File: tasks/mina/rails.rb (hunk 1 above)
Comment (the-spectator) [reply tasks/mina/rails.rb]: @lovro-bikic Applied suggested changes.

**Review by lovro-bikic — APPROVED** [review]
Thanks for

**Comment by lovro-bikic** [comment]
Reverted in https://github.com/mina-deploy/mina/pull/723, the main reason being backward compatibility.

This still doesn't prevent you from using Webpack with Mina on your project because you can adjust the variables in your project's `deploy.rb` the same way they have been adjusted in this PR:
```ruby
# config/deploy.rb
set :compiled_asset_path, ['public/assets', 'public/packs']
set :asset_dirs, ['vendor/assets/', 'app/assets/', 'app/javascript/', 'package.json']
set :shared_dirs, fetch(:shared_dirs, []).push('log', 'tmp/cache', 'node_modules', *fetch(:compiled_asset_path))
```

**Pull request merged**